content/docs/userspace/dist/glob.md

```diff
@@ -0,0 +1,102 @@
+
+The `desk.docket-0` file must include exactly one of the following clauses:
+
+#### `site+/some/path`
+
+If an app binds an Eyre endpoint and handles HTTP directly, for example with a [`%connect` task:eyre](/docs/arvo/eyre/tasks#connect), the `%site` clause is used, specifying the Eyre binding. In this case a glob is omitted entirely.
+
+#### `glob-ames+[~zod 0vs0me.h4sh]`
+
+If the glob is to be distributed over Ames, the `%glob-ames` clause is used, with a cell of the `ship` which has the glob and the `@uv` hash of the glob. If it's our ship, the hash can just be `0v0` and the glob can instead be created with the [Globulator](#globulator).
+
+#### `glob-http+['https://example.com/some.glob' 0vs0me.h4sh]`
+
+If the glob is to be distributed over HTTP, for example from an s3 instance, the `%glob-http` clause is used. It takes a cell of a `cord` with the URL serving the glob and the `@uv` hash of the glob.
+
+## Making a glob
+
+There are a couple of different methods depending on whether the glob will be distributed over HTTP or Ames.
+
+### Globulator
+
+For globs distributed over Ames from our ship, the client bundle can be uploaded directly with `%docket`'s Globulator tool, which is available in the browser at `http[s]://[host]/docket/upload`. It looks like this:
+
+![Globulator](https://media.urbit.org/docs/userspace/dist/globulator.png)
+
+Simply select the target desk, select the folder to be globulated, and hit `glob!`.
+
+Note the target desk must have been `|install`ed before uploading its glob. When installed, `%docket` will print `docket: awaiting manual glob for %desk-name desk` in the terminal and wait for the upload. The hash in the `%ames-glob` clause of the docket file will be overwritten by the hash of the new glob. As a result, there's no need to specify the actual glob hash in `desk.docket` - you can just use any `@uv` like `0v0`. Once uploaded, the desk can then be published with `:treaty|publish %desk-name` and the glob will become available for download by subscribers.
+
+### `-make-glob`
+
+There's a different process for globs to be distributed over HTTP from a webserver rather than over Ames from a ship. For this purpose, the `%garden` desk includes a `%make-glob` thread. The thread takes a folder in a desk and produces a glob of the files it contains, which it then saves to Unix in a [`jam`](/docs/hoon/reference/stdlib/2p#jam)file with a `.glob` extension.
+
+To begin, you'll need to spin up a fake ship and `|mount` a desk for which to add the files. In order for Clay to add the files, the desk must contain `mark` files in its `/mar` directory for all file extensions your folder contains. The `%garden` desk is a good bet because it includes `mark` files for `.js`, `.html`, `.png`, `.svg`, `.woff2` and a couple of others. If there's no desk with a mark for a particular file type you want included in your glob, you may need to add a new mark file. A very rudimentary mark file like the `png.hoon` mark will suffice.
```

Comment:
> To begin, you'll need to spin up a fake ship

Of course, this can also be done on a real ship, but you run into the clay bloat issue that's been warned about elsewhere.